index.js

```diff
@@ -76,5 +76,15 @@ export default async function pRetry(input, options) {
 				}
 			}
 		});
+
+		if (options.signal && !options.signal.aborted) {
+			options.signal.addEventListener('abort', () => {
+				operation.stop();
+				const reason = options.signal.reason ?? new DOMException('The operation was aborted.');
+				reject(reason instanceof Error ? reason : new DOMException(reason));
```

Comment:
Didn't realize `DOMException` was only added in Node 17. Should we use a plain `Error` instead?

Reply:
Use `DOMException` conditionally if available.